mmdet/models/dense_heads/detr_head.py

```diff
@@ -246,35 +132,40 @@ def forward_single(self, x: Tensor,
             head with normalized coordinate format (cx, cy, w, h). \
             Shape [nb_dec, bs, num_query, 4].
         """
-        # construct binary masks which used for the transformer.
-        # NOTE following the official DETR repo, non-zero values representing
-        # ignored positions, while zero values means valid positions.
-        batch_size = x.size(0)
-        input_img_h, input_img_w = batch_img_metas[0]['batch_input_shape']
-        masks = x.new_ones((batch_size, input_img_h, input_img_w))
-        for img_id in range(batch_size):
-            img_h, img_w, = batch_img_metas[img_id]['img_shape']
-            masks[img_id, :img_h, :img_w] = 0
-
-        x = self.input_proj(x)
-        # interpolate masks to have the same spatial shape with x
-        masks = F.interpolate(
-            masks.unsqueeze(1), size=x.shape[-2:]).to(torch.bool).squeeze(1)
-        # position encoding
-        pos_embed = self.positional_encoding(masks)  # [bs, embed_dim, h, w]
-        # outs_dec: [nb_dec, bs, num_query, embed_dim]
-        outs_dec, _ = self.transformer(x, masks, self.query_embedding.weight,
-                                       pos_embed)
-
         all_cls_scores = self.fc_cls(outs_dec)
         all_bbox_preds = self.fc_reg(self.activate(
             self.reg_ffn(outs_dec))).sigmoid()
         return all_cls_scores, all_bbox_preds
 
+    def loss(self, x: Tuple[Tensor], batch_data_samples: SampleList) -> dict:
+        """Perform forward propagation and loss calculation of the detection
+        head on the features of the upstream network.
+
+        Args:
+            x (tuple[Tensor]): Features from the transformer detector, each is
```

Comment:
Features from the transformer decoder?
Please check its type Seem it should be a tensor with shape (num_layers, Num_queries, batch_size, Embed_dim) or 
 (num_layers,batch_size, Num_queries, Embed_dim)

Reply:
Resolved.